### PR TITLE
Supporting pep 0518

### PR DESCRIPTION
**Pull request by renefritze**
Pip 18 supports PEP 518. Question is if we want this merged for 0.5 or later.

### CODE REVIEW

**Comment by sdrave** [comment]
@renefritze, what would be the consequences for older PIPs?

**Comment by renefritze** [comment]
None. They will still try to execute setup.py and see the setup_requires. And then maybe fail if they're old enough.

**Comment by renefritze** [comment]
There are however consequences for current Pips, so we decided to defer this.

**Comment by renefritze** [comment]
I've looked into the many open issues and  PRs in docopt and it seems pretty much abandoned at this point. In light of #540 I propose to include docopt (MIT licensed, only used in demos) in pyMOR and removed it again after the transition to notebooks.

File: src/pymor/tools/LICENSE-MIT [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY
+KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+(This file is included for docopt.py only)
```
Comment (sdrave) [inline src/pymor/tools/LICENSE-MIT]: Maybe just put it in the header of docopt.py?

**Comment by renefritze** [comment]
@pymor/pymor-devs I've just (re-)enabled the check requirements for GitLab and Azure (both Windows and OSX)

**Pull request merged**